scripts/assignments_test.py

```diff
@@ -0,0 +1,36 @@
+import tempfile, os, re
+from git import Repo
+from shutil import copyfile
+
+def get_directories(basedir):
+    return [os.path.join(basedir, dir) for dir in os.listdir(basedir) \
+                                       if os.path.isdir(os.path.join(basedir, dir)) \
+                                       and not dir.startswith('.')]
+
+pipeline_failed = False
+with tempfile.TemporaryDirectory() as tempdir:
+    # Clone the assignments repository in a temporary directory.
+    Repo.clone_from('https://github.com/cse1110/assignments', tempdir)
+    print('Assignments repository cloned.')
+
+    for category_dir in get_directories(tempdir):
+        for assignment_dir in get_directories(category_dir):
+            # Change to the assignment directory.
+            os.chdir(assignment_dir)
+
+            # Copy the solution to the correct folder.
+            testfile = os.listdir(os.path.join(assignment_dir, 'solution'))[0]
+            copyfile(f'{assignment_dir}/solution/{testfile}', f'{assignment_dir}/src/test/java/delft/{testfile}')
+
+            # Run `andy` on the current assignment.
+            output = os.popen('mvn andy:run').read()
+            score = int(re.search('Final grade: [0-9]+', output).group().split(' ')[2])
```

Comment:
Maybe you could add a check to verify the correct Andy version is used - Andy prints the hash of the commit that it was built on in the output it generates.